Public/Src/Engine/Dll/Engine.cs

```diff
@@ -2452,6 +2459,16 @@ private static bool IsJournalPossiblyAvailableForPath(PathTable pathTable, Absol
                     if (!possibleIdentity.Succeeded &&
                         possibleIdentity.Failure.Content == VersionedFileIdentity.IdentityUnavailabilityReason.NotSupported)
                     {
+                        try
+                        {
+                            string finalPathStr = FileUtilities.GetFinalPathNameByHandle(handle);
+                            finalPath = AbsolutePath.Create(pathTable, finalPathStr);
+                        }
+                        catch (NativeWin32Exception)
+                        {
+                            finalPath = path;
+                        }

```

Comment:
Is this legit?

Reply:
It is legit because GetFinalPathNameByHandle can throw NativeWin32Exception.

---
In reply to: [344283054](https://github.com/microsoft/BuildXL/pull/1183#discussion_r344283054) [](ancestors = 344283054)